openlibrary/components/BulkSearch/utils/classes.js

```diff
@@ -8,13 +8,82 @@ export class ExtractedBook {
     }
 }
 
+class AbstractExtractor {
+    constructor(name = '') {
+        this.name = name
+    }
+    async run() {
+    }
+}
+
+export class RegexExtractor extends AbstractExtractor {
+    constructor(name = '', pattern = ''){
+        super(name)
+        this.pattern = new RegExp(pattern, 'gmu');
+    }
+    async run(_extractOptions, text = '') {
+        const data = [...text.matchAll(this.pattern)]
+        const extractedBooks = data.map((entry) => new ExtractedBook(entry.groups?.title, entry.groups?.author))
+        const matchedBooks = extractedBooks.map((entry) => new BookMatch(entry, []))
+        return matchedBooks
+    }
+}
+
+export class AIExtractor extends AbstractExtractor{
+
+    constructor(name = '', model = '') {
+        super(name)
+        this.model = model
+    }
+    async run(extractOptions, text= '') {
```

Comment:
Type hints here please! And on the other extractor class/abstract class.

Reply:
Resolved